integration/solana/simple_collectible.sol

```diff
@@ -0,0 +1,105 @@
+
+    // The mint account will identify the NFT in this example
+    constructor (address _mintAccount, address _metadataAuthority) {
+        mintAccount = _mintAccount;
+        metadataAuthority = _metadataAuthority;
+    }
+
+    /// Create a new NFT and associate it to an URI
+    ///
+    /// @param tokenURI a URI that leads to the NFT resource
+    /// @param mintAuthority an account that signs each new mint
+    /// @param ownerTokenAccount the owner associated token account
+    function createCollectible(string memory tokenURI, address mintAuthority, address ownerTokenAccount) public {
+        SplToken.TokenAccountData token_data = SplToken.get_token_account_data(ownerTokenAccount);
+
+        // The mint will only work if the associated token account points to the mint account in this contract
+        // This assert is not necessary. The transaction will fail if this does not hold true.
+        assert(mintAccount == token_data.mintAccount);
+        SplToken.MintAccountData mint_data = SplToken.get_mint_account_data(token_data.mintAccount);
+        // Ensure the supply is zero. Otherwise, this is not an NFT.
+        assert(mint_data.supply == 0);
+        
+        // An NFT on Solana is a SPL-Token with only one minted token.
+        // The token account saves the owner of the tokens minted with the mint account, the respective mint account and the number
+        // of tokens the owner account owns
+        SplToken.mint_to(token_data.mintAccount, ownerTokenAccount, mintAuthority, 1);
+        updateNftUri(tokenURI);
+        // Log on blockchain records information about the created token
+        emit NFTMinted(token_data.owner, token_data.mintAccount);
+    }
+
+    /// Transfer ownership of this NFT from one account to another
+    ///
+    /// @param oldOwnerTokenAccount the associated token account for the current owner
+    /// @param newOwnerTokenAccount the associated token account for the new owner
+    function transferOwnership(address oldOwnerTokenAccount, address newOwnerTokenAccount) public {
+        // The current owner does not need to be the caller of this functions, but they need to sign the transaction
+        // with their private key.
+        SplToken.TokenAccountData old_data = SplToken.get_token_account_data(oldOwnerTokenAccount);
+        SplToken.TokenAccountData new_data = SplToken.get_token_account_data(newOwnerTokenAccount);
+
+        // Ensure we are dealing with the token this contract refers to
+        assert(old_data.mintAccount == mintAccount);
+        assert(new_data.mintAccount == mintAccount);
+
+        // To transfer the ownership of a token, we need the current owner and the new owner. The payer account is the account used to derive
+        // the correspondent token account in TypeScript.
+        SplToken.transfer(oldOwnerTokenAccount, newOwnerTokenAccount, old_data.owner, 1);
+        emit NFTSold(old_data.owner, new_data.owner);
+    }
```

Comment:
For the purposes of education, you should specify that this is just wrapping the innate SPL transfer, and that the value of this instruction is to emit the `NFTSold` event.